clients/roleProviders/elrondRoleProvider_test.go

```diff
@@ -0,0 +1,104 @@
+package roleProviders
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-eth-bridge/testsCommon/bridgeV2"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+	"github.com/stretchr/testify/assert"
+)
+
+func createElrondMockArgs() ArgsElrondRoleProvider {
+	return ArgsElrondRoleProvider{
+		Log:        logger.GetOrCreate("test"),
+		DataGetter: &bridgeV2.DataGetterStub{},
+	}
+}
+
+func TestNewElrondRoleProvider(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil data getter should error", func(t *testing.T) {
+		args := createElrondMockArgs()
+		args.DataGetter = nil
+
+		erp, err := NewElrondRoleProvider(args)
+		assert.True(t, check.IfNil(erp))
```

Comment:
why not `assert.Nil(t, erp)` ?

Reply:
Through this, I've also checked the IsInterfaceNil works properly :)

Reply:
correct. nice